packages/react-storage/src/components/StorageBrowser/__tests__/createStorageBrowser.spec.tsx

```diff
@@ -12,9 +13,30 @@ const config = {
 };
 
 describe('createStorageBrowser', () => {
-  it('returns a StorageBrowser', () => {
+  it('returns a StorageBrowser', async () => {
     const { StorageBrowser } = createStorageBrowser({ config });
 
-    expect(render(<StorageBrowser />).container).toBeDefined();
+    await waitFor(() => {
+      expect(render(<StorageBrowser />).container).toBeDefined();
+    });
+  });
+
+  it('renders ActionView on action select', async () => {
+    const { StorageBrowser } = createStorageBrowser({ config });
+
+    // Used 'mockReturnValueOnce` three times to mock each time it's called
+    // to render LocationActionView
+    // Twice in `DefaultStorageBrowser` and then once in `LocationActionView`
+    jest
+      .spyOn(controlsModule, 'useControl')
+      .mockReturnValueOnce([{ location: {} }])
+      .mockReturnValueOnce([{ selected: { actionType: 'CREATE_FOLDER' } }])
+      .mockReturnValueOnce([{ selected: { actionType: 'CREATE_FOLDER' } }]);
```

Comment:
this seems really fragile, does anyone have thoughts on how to make this better? The coverage was off by .04% and I just wanted to get it through without having to lower the coverage for now

Reply:
There's no need to complicate the logic here by considering child component behavior since `location` does not apply in the case of `ActionView` rendering, can make the assertion against:
```suggestion
 jest
      .spyOn(controlsModule, 'useControl')
      .mockReturnValueOnce([{ selected: { actionType: 'CREATE_FOLDER' } }])
```